query

```diff
@@ -1,25 +1,34 @@
 #!/bin/bash
-# Convenience script to query for a similar files for a single file
+# Convenience script to query for a similar files of a given file
 
 E_BUILD_FAILED=142
 
 version=$(grep "version :=" build.sbt | cut -d'=' -f 2 | tr -d ' ' | tr -d '"')
-jar="target/gemini-${version}.jar"
-build_command="./sbt package"
-current_dir="$(dirname "$0")"
+jar="target/gemini-uber.jar"
+deps_jar="target/gemini-assembly-${version}-deps.jar"
+
+build_command="./sbt assembly"
+build_deps_command="./sbt assemblyPackageDependency"
 
-app_class="tech.sourced.gemini.QueryApp"
+current_dir="$(dirname "$0")"
+app_class="tech.sourced.gemini.cmd.QueryApp"
 
 hash java >/dev/null 2>&1 || { echo "Please install Java" >&2; exit 1; }
 
+if [[ ! -f "${deps_jar}" ]]; then
+    echo "${deps_jar} not found. Running build '${build_deps_command}'"
+    if ! $build_deps_command ; then
+        exit "${E_BUILD_FAILED}"
+    fi
+fi
+
 if [[ ! -f "${jar}" ]]; then
     echo "${jar} not found. Running build '${build_command}'"
     if ! $build_command ; then
         exit "${E_BUILD_FAILED}"
     fi
 fi
 
-source "$current_dir/scripts/common_params.sh"
+source "${current_dir}/scripts/common_params.sh"
 
-#exec java -cp "${jar}" "${app_class}" $@
-exec ./sbt --warn "run-main ${app_class} ${COMMAND_PARAMS[*]}"
+exec java -cp "${jar}:${deps_jar}" "${app_class}" $@
```

Comment:
I would like to still be able to run `query` using sbt for development. It would be nice to read ENV variable and depend on it run `sbt` or `java`.
Otherwise currently for any change of source code developers must run 2-3 commands which work much slower than `./sbt package`.
What do you think?

Reply:
I did not understood your suggestion, but from what I see right now - it's only `./sbt assembly` that is needed to be re-run on any change in code. 

`./sbt assemblyPackageDependency` would only be needed if new dependency is added.

Such developer experience looks good to me, that being said, I'm open to suggestions. 
Please, feel free to propose better developer experience in a new issue with desired behaviour and we can work from there.

Reply:
before `./query -params ...` would recompile and run the query command. It was very handy and for me, as a developer, it's really important to keep some way to run&execute.

Reply:
As discussed offline, the suggestion was to keep best of both worlds:
 - for development of non-Spark tools: keep single command that builds and runs (`./sbt run-main`) 
 - otherwise, use `build & java -cp` way of running, same for all 3 tools

Reply:
@smacker Would something like 50b52a9 make sense to you? Addressed by adding "DEV" env var.

Personally, although I appreciate interpreted languages workflow, I do not like this: those are two very different ways to organize classpath, and thus it opens the possibility of "everything works on my machine" for developer, who used `./sbt`.

But CI works on final .jars so this hopefully will be catched early.